### PR TITLE
add new operations and mark db and collection names as optional

### PR DESCRIPTION
**Pull request by MongoCaleb**
### Jira

https://jira.mongodb.org/browse/DOCSP-33724

### Staged Changes

https://preview-mongodbmongocaleb.gatsbyjs.io/atlas-app-services/docsp-33724/triggers/database-triggers/#database-change-events
https://preview-mongodbmongocaleb.gatsbyjs.io/atlas-app-services/docsp-33724/reference/config/triggers/#database-triggers

### Reminder Checklist

You might need to also update some corresponding pages. Check if completed or N/A.

- [ ] Create Jira ticket for corresponding docs-realm update(s), if any
- [ ] Checked/updated Admin API
- [ ] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-app-services/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/triggers/database-triggers.txt [hunk 1]
```diff
@@ -287,6 +287,35 @@ following operation types:
...
+   * - **Drop_Collection**
+     - Represents a collection being dropped.
+
+   * - **Shard_Collection**
+     - Represents a collection getting sharded.
+
+   * - **Reshard_Collection**
+     - Represents a change to a collection's sharding.
+
+   * - **Refine_Collection_Shard_Key**
```
Comment (nathantfrank) [inline source/triggers/database-triggers.txt:308]: Not sure if this is possible, to update the spacing on this table as the `y` in this line is causing a new line.

File: source/triggers/database-triggers.txt [hunk 2]
```diff
@@ -287,6 +287,35 @@ following operation types:
...
+     - Represents the modification collection.
+
+   * - **Rename_Collection**
+     - Represents collection being renamed.
+
+   * - **Drop_Collection**
+     - Represents a collection being dropped.
+
+   * - **Shard_Collection**
+     - Represents a collection getting sharded.
```
Comment (nlarew) [inline source/triggers/database-triggers.txt:303]:
```suggestion
     - Represents a collection changing from unsharded to sharded.
```

**Review by nlarew — APPROVED** [review]
LGTM % one change

File: source/reference/config/triggers.txt [hunk 3]
```diff
@@ -155,6 +156,16 @@ configuration files:
...
        - ``"DELETE"``
+       - ``"CREATE_COLLECTION"``
+       - ``"MODIFY_COLLECTION"``
+       - ``"RENAME_COLLECTION"``
+       - ``"SHARD_COLLECTION"``
+       - ``"DROP_COLLECTION"``
+       - ``"RESHARD_COLLECTION"``
+       - ``"REFINE_COLLECTION_SHARD_KEY"``
+       - ``"CREATE_INDEXES"``
+       - ``"DROP_INDEXES"``
```
Comment (nathantfrank) [inline source/reference/config/triggers.txt:167]: [Q] do we want to add a blurb that these are `Database and Deployment only` won't fire/unsupported if watching at the collection level.

File: source/reference/config/triggers.txt [hunk 4]
```diff
@@ -155,6 +156,16 @@ configuration files:
...
+       - ``"DROP_DATABASE"``
```
Comment (nathantfrank) [inline source/reference/config/triggers.txt:168]: [q] similar to above supported for Deployment only (db and coll are blank)

**Comment by MongoCaleb** [comment]
Closing -- merged into PR #611

**Pull request closed without merging**